drivers/pipes/pipe_common.h

```diff
@@ -117,6 +117,8 @@ struct pipe_dev_s
   mutex_t    d_bflock;      /* Used to serialize access to d_buffer and indices */
   sem_t      d_rdsem;       /* Empty buffer - Reader waits for data write */
   sem_t      d_wrsem;       /* Full buffer - Writer waits for data read */
+  sem_t      d_nrdsem;      /* Waits for readers */
+  sem_t      d_nwrsem;      /* Waits for writers */
```

Comment:
Where those semaphores are initialized?

Reply:
I'm sorry! It may have been lost during one of my rebase.